src/headers.zig

```diff
@@ -38,20 +33,17 @@ pub const Headers = struct {
     }
 
     pub fn appendValue(self: *Headers, name: []const u8, value: []const u8) !void {
-        const duped_name = try self.allocator.dupe(u8, name);
-        const duped_value = try self.allocator.dupe(u8, value);
-
-        try self.list.append(.{
-            .name = duped_name,
-            .value = duped_value,
+        try self.append(.{
+            .name = name,
+            .value = value,
         });
     }
 
-    pub inline fn append(self: *Headers, header: Header) !void {
-        return self.appendValue(header.name, header.value);
+    pub fn append(self: *Headers, header: Header) !void {
+        return self.list.append(header.name, header.value);
     }
 
-    pub inline fn appendSlice(self: *Headers, headers: HeadersSlice) !void {
+    pub fn appendSlice(self: *Headers, headers: HeadersSlice) !void {
```

Comment:
These changes require *every* header in `headers` to remain alive for the entire life of this object, which is now the caller's responsibility and is not guaranteed, this should be noted.